src/Mod/Sketcher/Gui/DrawSketchHandlerBSplineByInterpolation.h

```diff
@@ -0,0 +1,606 @@
+        BSplineKnots.clear();
+        BSplineMults.clear();
+
+        eraseEditCurve();
+
+        addSugConstraint();
+
+        IsClosed = false;
+    }
+
+    QString getCrosshairCursorSVGName() const override
+    {
+        return QString::fromLatin1("Sketcher_Pointer_Create_BSpline");
+    }
+
+    void addSugConstraint()
+    {
+        std::vector<AutoConstraint> sugConstr1;
+        sugConstr.push_back(std::move(sugConstr1));
+    }
+
+    // NOTE: In this context, it is not a control polygon, but a 1-degree interpolation
+    void drawControlPolygonToPosition(Base::Vector2d position)
+    {
+
+        std::vector<Base::Vector2d> editcurve(BSplineKnots);
+        editcurve.push_back(position);
+
+        drawEdit(editcurve);
+    }
+
+    void drawCursorToPosition(Base::Vector2d position)
+    {
+        if (!BSplineKnots.empty()) {
+            float length = (position - BSplineKnots.back()).Length();
+            float angle = (position - BSplineKnots.back()).GetAngle(Base::Vector2d(1.f,0.f));
+
+            if (showCursorCoords()) {
+                SbString text;
+                std::string lengthString = lengthToDisplayFormat(length, 1);
+                std::string angleString = angleToDisplayFormat((angle != -FLOAT_MAX) ? angle * 180 / M_PI : 0, 1);
+                text.sprintf(" (%s, %s)", lengthString.c_str(), angleString.c_str());
+                setPositionText(position, text);
+            }
+        }
+    }
+
+    void eraseEditCurve()
+    {
+        drawEdit(std::vector<Base::Vector2d>());
+    }
+
+    bool finishCommand(Base::Vector2d position)
+    {
+        if (Mode==STATUS_CLOSE) {
+            // TODO: make modifications for construction by knots
+            unsetCursor();
+            resetPositionText();
+
+            unsigned int myDegree = 3;
+
+            if (ConstrMethod == 0) {
+                BSplineMults.front() = myDegree + 1; // FIXME: This is hardcoded until degree can be changed
+                BSplineMults.back() = myDegree + 1; // FIXME: This is hardcoded until degree can be changed
+            }
+
+            std::vector<std::stringstream> streams;
+
+            // The first point
+            streams.emplace_back();
+            streams.back() << "App.Vector(" << BSplineKnots.front().x << "," << BSplineKnots.front().y << "),";
+            // Middle points
+            for (size_t i = 1; i < BSplineKnots.size() - 1; ++i) {
+                streams.back() << "App.Vector(" << BSplineKnots[i].x << "," << BSplineKnots[i].y << "),";
+                if (BSplineMults[i] >= myDegree) {
+                    streams.emplace_back();
+                    streams.back() << "App.Vector(" << BSplineKnots[i].x << "," << BSplineKnots[i].y << "),";
+                }
+            }
+            // The last point
+            streams.back() << "App.Vector(" << BSplineKnots.back().x << "," << BSplineKnots.back().y << "),";
+
+            std::vector<std::string> controlpointses; // Gollum gollum

```

Comment:
Hope Smeagol does not jump on me. Any purpose for that comment?

Reply:
It does remind the reader that this this is a vector of strings that represent lists. I can be more explicit here.

Reply:
It must be a reference to something I ignore. Gollum gollum says nothing to me. If I try hard I can imagine the array trying to rule all other arrays in FreeCAD.

Reply:
Just that Gollum often makes plural of plurals, "ringses".

Reply:
Please, think of the our dear new Developer from Antarctica that is not familiar with this kind of references. She is just down there at -60 Celsius  trying to understand what you wrote ;)

Reply:
OK I'll change it. Who is this legend though?

Reply:
Didn't you meet her yet? LoL

Reply:
Not really.

Reply:
She is very friendly and a real team player. She know her c++ too. I am sure you will like her.

Reply:
Introduce us, then. Half expecting this person to actually be a penguin and good at sea++.